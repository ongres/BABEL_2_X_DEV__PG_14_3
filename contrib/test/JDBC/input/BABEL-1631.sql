-- Check if exact input typmod is retrieved on TDS side for char/nchar datatypes
-- when engine provided typmod = -1

DROP FUNCTION IF EXISTS custom_f3;
go
CREATE FUNCTION custom_f3(@one CHAR(10)) RETURNS CHAR(6) AS BEGIN RETURN @one; END;
go
SELECT custom_f3('abcdef');
go

DROP FUNCTION IF EXISTS custom_f4;
go
CREATE FUNCTION custom_f4(@one NCHAR(10)) RETURNS NCHAR(6) AS BEGIN RETURN @one; END;
go
SELECT custom_f4('abc');
go

DROP FUNCTION IF EXISTS custom_f5;
go
CREATE FUNCTION custom_f5(@one CHAR(20)) RETURNS CHAR(8) AS BEGIN RETURN @one; END;
go
SELECT custom_f5('abcdefghij');
go

DROP FUNCTION IF EXISTS custom_f6;
go
CREATE FUNCTION custom_f6(@one CHAR(10)) RETURNS CHAR AS BEGIN RETURN @one; END;
go
SELECT custom_f6('abcdef');
go

DROP FUNCTION custom_f3
DROP FUNCTION custom_f4
DROP FUNCTION custom_f5
DROP FUNCTION custom_f6
GO
