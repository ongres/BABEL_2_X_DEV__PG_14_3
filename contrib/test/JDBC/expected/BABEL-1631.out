
-- Check if exact input typmod is retrieved on TDS side for char/nchar datatypes
-- when engine provided typmod = -1
DROP FUNCTION IF EXISTS custom_f3;
go
CREATE FUNCTION custom_f3(@one CHAR(10)) RETURNS CHAR(6) AS BEGIN RETURN @one; END;
go
SELECT custom_f3('abcdef');
go
~~START~~
char
abcdef
~~END~~


DROP FUNCTION IF EXISTS custom_f4;
go
CREATE FUNCTION custom_f4(@one NCHAR(10)) RETURNS NCHAR(6) AS BEGIN RETURN @one; END;
go
SELECT custom_f4('abc');
go
~~START~~
nchar
abc   
~~END~~


DROP FUNCTION IF EXISTS custom_f5;
go
CREATE FUNCTION custom_f5(@one CHAR(20)) RETURNS CHAR(8) AS BEGIN RETURN @one; END;
go
SELECT custom_f5('abcdefghij');
go
~~START~~
char
abcdefgh
~~END~~


DROP FUNCTION IF EXISTS custom_f6;
go
CREATE FUNCTION custom_f6(@one CHAR(10)) RETURNS CHAR AS BEGIN RETURN @one; END;
go
SELECT custom_f6('abcdef');
go
~~START~~
char
a
~~END~~


DROP FUNCTION custom_f3
DROP FUNCTION custom_f4
DROP FUNCTION custom_f5
DROP FUNCTION custom_f6
GO
